src/views/users/User/UserCreateUpdate.vue

```diff
@@ -231,11 +233,26 @@ export default {
       this.initial.org_roles = roles.filter(role => role.name === 'OrgUser').map(role => role.id)
     },
     disableMFAFieldIfNeed(user) {
+      let options = null
+      let mfa_level = null
       // SECURITY_MFA_AUTH 0 不开启 1 全局开启 2 管理员开启
+      const securityMFAAuth = store.getters.publicSettings['SECURITY_MFA_AUTH']
       const adminUserIsNeed = (user?.is_superuser || user?.is_org_admin) && this.$route.meta.action === 'update' &&
-        store.getters.publicSettings['SECURITY_MFA_AUTH'] === 2
-      if (store.getters.publicSettings['SECURITY_MFA_AUTH'] === 1 || adminUserIsNeed) {
-        this.fieldsMeta['mfa_level'].disabled = true
+        securityMFAAuth === MFASystemSetting.onlyAdminUsers
+      if (securityMFAAuth === MFASystemSetting.allUsers) {
+        options = [{ 'value': MFALevel.allUsers, 'label': this.$t('MFAAllUsers') }]
+        mfa_level = MFALevel.allUsers
+      }
+      if (securityMFAAuth === MFASystemSetting.onlyAdminUsers && adminUserIsNeed) {
+        options = [{
+          'value': MFALevel.onlyAdminUsers,
+          'label': this.$t('MFAOnlyAdminUsers')
+        }]
+        mfa_level = MFALevel.onlyAdminUsers
+      }
+      if (mfa_level !== null && options !== null) {
+        this.fieldsMeta['mfa_level'].options = options
+        this.initial.mfa_level = mfa_level
       }
     }
   }
```

Comment:
There aren't major discrepancies, but there may be some small inconsistencies due to the difference in version numbers between 2021 and 2024.

Potential improvements could include optimizing rendering of elements like Select2 and adding a few more features or fixing other minor bugs. However, these improvements would need thorough testing before implementation.

This code is already very clean and follows good software development standards, so most changes are best left for maintenance purposes rather than redesigns based on current practices.